test/specs/services/faye.spec.js

```diff
@@ -32,26 +31,61 @@ const sandbox = sinon.sandbox.create();
 
 describe('Faye service', () => {
     let mockedStore;
+    let getMessagesStub;
+    let disconnectFayeStub;
+    let handleConversationUpdatedStub;
+    let showSettingsStub;
+    let hideChannelPageSpy;
+    let hideConnectNotificationSpy;
+    let addMessageSpy;
+    let incrementUnreadCountSpy;
+    let resetUnreadCountSpy;
+    let setUserSpy;
+    let setFayeConversationSubscriptionSpy;
+    let setFayeUserSubscriptionSpy;
 
     beforeEach(() => {
         mockedStore = createMockedStore(sandbox, getProps());
 
         sandbox.stub(Client.prototype, 'addExtension');
-        sandbox.stub(conversationService, 'getMessages');
-        sandbox.stub(conversationService, 'disconnectFaye');
-        sandbox.stub(conversationService, 'handleConversationUpdated');
-        sandbox.stub(appService, 'showSettings');
-        sandbox.stub(appService, 'hideChannelPage');
-        sandbox.stub(appService, 'hideConnectNotification');
-
-        sandbox.stub(conversationActions, 'addMessage');
-        sandbox.stub(conversationActions, 'incrementUnreadCount');
-        sandbox.stub(conversationActions, 'resetUnreadCount');
-        sandbox.stub(userActions, 'setUser');
-        sandbox.stub(fayeActions, 'setFayeConversationSubscription');
-        sandbox.stub(fayeActions, 'setFayeUserSubscription');
-
-        sandbox.stub(utilsDevice, 'getDeviceId').returns(123);
+
+        getMessagesStub = sandbox.stub().returns(() => Promise.resolve());
+        FayeRewire('getMessages', getMessagesStub);
```

Comment:
So let me see if I understand what's happening here, because I was a bit confused at first glance. 

We're testing faye in this file
The faye module imports a bunch of symbols from other places
`FayeRewire` is a sort of dictionary of all the symbols imported by the Faye module
In this file, you import the same symbols that Faye is importing, but then hot-swap the reference _inside the faye module_ with a stubbed version?

So it's not working like a traditional mock where you replace the implementation of a function, what you're really replacing is the faye module's internal reference to that function. Is that right?

Basically I was confused how you could change stubs on different modules to instead all point to `FayeRewire`. I was expecting `ConversationActionsRewire` + `UserActionsRewire` + `ConversationServiceRewire`, etc.